src/app/modules/map/components/map/map.component.ts

```diff
@@ -767,37 +772,29 @@ export class MapComponent implements OnInit {
   /**
    * Listen for click events on graphics on a specific layer.
    * @param layer the layer of interest
-   * @return a stream of click events. Contains the graphics that was hit. If no hit, the event will contain an undefined object
+   * @return a stream of click events. Contains the graphics that was hit. If no hit, the event will contain null
    */
-  createClickEventHandler(layer: Layer): Observable<Graphic|undefined> {
-    this.view.popup.autoOpenEnabled = false;
-    this.clickSubscriptions.push(this.view.on('click', (event) => {
-      const screenPoint = {
-        x: event.x,
-        y: event.y
-      };
-      // Search for graphics at the clicked location
-      this.view
-        .hitTest(screenPoint, { include: [layer] })
-        .then((response) => {
-          if (response.results.length) {
-            const graphic = response.results.filter((result) => {
-              return result.graphic.layer === layer; // check if the graphic belongs to the layer of interest
-            })[0].graphic;
-            this.clickEvent.next(graphic); //we hit a graphic on this layer
-          } else {
-            this.clickEvent.next(undefined); //we didn't hit anything
-          }
-        })
-        .catch((err) => {
-          this.logger.error(
-            err,
-            DEBUG_TAG,
-            `Click event failed: ${err}`
-          );
-        });
-    }));
-    return this.clickEvent.asObservable();
+  createClickHandlerForGraphicsLayer(layer: GraphicsLayer): Observable<Graphic|null> {
+    const getGraphicsAtScreenPoint = p => this.view
+      .hitTest(p, { include: layer })
+      .catch(err => {
+        this.logger.error(err, DEBUG_TAG, 'hitTest failed');
+        return null;
+      });
+
+    const getFirstGraphicOrNull = (response: __esri.HitTestResult) => {
+      try {
+        const [firstHit] = response.results;
+        return firstHit.graphic;
```

Comment:
Denne ryker på en exception hvis du ikke får treff.
Skulle ikke heller testet for om den var undefined?
Jeg har vært vant til å unngå bruk av exceptions hvis det ikke skjer en feil, så man ikke risikerer at man maskerer andre uventede feil som kan dukke opp. Men er det vanlig å gjøre det slik i Javascript?

Reply:
Vet ikke helt hva som er best. Vi logger jo feil i hitTest på linje 781, så jeg tenkte dette var greit nok. Kan godt heller gjøre noe a la dette her?
```
if (response && response.result && response.result.length)
```
feks?

Reply:
Ja, men den destrucure-saken var jo kul!
Hva med: 
      ```
const [firstHit] = response.results;
      if(firstHit) {
        return firstHit.graphic;
      }
      return null;

```

Reply:
Siden dette uansett er inni en try...catch så kan det forsåvidt forenkles til én linje:
```
try {
  return response.results[0].graphic
}
```

Reply:
Kodeforslaget mitt var ikke i en try-catch-blokk. Det var nettopp try-catch-blokka jeg ikke likte:)